src/main/scala/org/apache/spark/shuffle/ucx/NvkvShuffleMapOutputWriter.scala

```diff
@@ -0,0 +1,291 @@
+    // TODO - need to find a better way yo commit blocks locations to DPU
+    // Idealy we will want to send all the locations at the end of the write stage
+    // and not during the write stage with samllest number of AM messages (1) as possible.
+    // The problem is how we can tell that we reached the last map partition.
+    CommonUtils.safePolling(() => {ucxTransport.progress()}, () => {!commitBlock})
+    partitionLengths
+  }
+
+  @throws[IOException]
+  override def abort(error: Throwable): Unit = {
+    cleanUp()
+  }
+
+  @throws[IOException]
+  private def cleanUp(): Unit = {
+    NvkvShuffleMapOutputWriter.log.debug("NvkvShuffleMapOutputWriter cleanUp")
+  }
+
+  @throws[IOException]
+  private def initStream(): Unit = {
+    NvkvShuffleMapOutputWriter.log.debug(s"NvkvShuffleMapOutputWriter initStream bufferSize $bufferSize")
+  }
+
+  @throws[IOException]
+  private def initChannel(): Unit = {
+    NvkvShuffleMapOutputWriter.log.debug("NvkvShuffleMapOutputWriter initChannel")
+    // This file needs to opened in append mode in order to work around a Linux kernel bug that
+    // affects transferTo; see SPARK-3948 for more details.
+  }
+
+  private class NvkvShufflePartitionWriter (private val partitionId: Int) extends ShufflePartitionWriter {
+    NvkvShuffleMapOutputWriter.log.debug("NvkvShufflePartitionWriter ctor " + partitionId + " shuffleId " + shuffleId + " mapId " + mapId)
+    private var partStream: PartitionWriterStream = null
+    private var partChannel: PartitionWriterChannel = null
+
+    @throws[IOException]
+    override def openStream: OutputStream = {
+      NvkvShuffleMapOutputWriter.log.debug("NvkvShufflePartitionWriter openStream " + partitionId)
+      if (partStream == null) {
+        initStream()
+        partStream = new PartitionWriterStream(partitionId)
+      }
+      partStream
+    }
+
+    @throws[IOException]
+    override def openChannelWrapper: Optional[WritableByteChannelWrapper] = {
+      NvkvShuffleMapOutputWriter.log.debug("NvkvShufflePartitionWriter openChannelWrapper " + partitionId)
+      if (partChannel == null) {
+        if (partStream != null) throw new IllegalStateException("Requested an output stream for a previous write but" + " now an output channel has been requested. Should not be using both channels" + " and streams to write.")
+        initChannel()
+        partChannel = new PartitionWriterChannel(partitionId)
+      }
+      Optional.of(partChannel)
+    }
+
+    override def getNumBytesWritten: Long = {
+      NvkvShuffleMapOutputWriter.log.debug("NvkvShufflePartitionWriter getNumBytesWritten " + partitionId)
+      if (partChannel != null) try partChannel.getCount
+      catch {
+        case e: IOException =>
+          throw new RuntimeException(e)
+      }
+      else if (partStream != null) partStream.getCount
+      else {
+        // Assume an empty partition if stream and channel are never created
+        0
+      }
```

Comment:
```suggestion
      else if (partStream != null) partStream.getCount else 0
```
or
```suggestion
      else 
      (partStream != null) ? partStream.getCount | 0
```

Reply:
IMO it's not a good practice to write if statements without curly braces (if we consider the maintainability of the code).
I've added braces to his statement, I think it's better to add braces to all statements.

Reply:
I agree, but the else if statement at line 222 didn't have braces and then 223 did, so I thought we can just one-line it

Reply:
Yes. previous statements were "bad habit". I've changed them as well